diff --git a/node_modules/inline-loops.macro/index.d.ts b/node_modules/inline-loops.macro/index.d.ts
index c3eeb39..19161b4 100644
--- a/node_modules/inline-loops.macro/index.d.ts
+++ b/node_modules/inline-loops.macro/index.d.ts
@@ -6,7 +6,7 @@ type Dictionary<Type> = {
 /**
  * iterables
  */
-type ArrayIterable = any[];
+type ArrayIterable = any[] | readonly any[];
 type ObjectIterable = Dictionary<any>;
 
 /**
